src/main/diagnostics/applicablityActionProvider.ts

```diff
@@ -0,0 +1,87 @@
+import * as vscode from 'vscode';
+import { CodeFileTreeNode } from '../treeDataProviders/issuesTree/codeFileTree/codeFileTreeNode';
+import { CodeIssueTreeNode } from '../treeDataProviders/issuesTree/codeFileTree/codeIssueTreeNode';
+import { FileTreeNode } from '../treeDataProviders/issuesTree/fileTreeNode';
+import { IssueTreeNode } from '../treeDataProviders/issuesTree/issueTreeNode';
+import { SeverityUtils } from '../types/severity';
+
+import { AbstractFileActionProvider } from './abstractFileActionProvider';
+
+export class ApplicablityActionProvider extends AbstractFileActionProvider implements vscode.CodeActionProvider {
+    /** @Override */
+    public activate(context: vscode.ExtensionContext) {
+        super.activate(context);
+        context.subscriptions.push(
+            this,
+            vscode.languages.registerCodeActionsProvider({ scheme: 'file', pattern: '**/*.{py,js}' }, this, {
+                providedCodeActionKinds: [vscode.CodeActionKind.Empty]
+            })
+        );
+    }
+
+    /** @Override */
+    provideCodeActions(
+        document: vscode.TextDocument,
+        range: vscode.Range | vscode.Selection,
+        context: vscode.CodeActionContext
+    ): vscode.Command[] | undefined {
+        // Search if the file had issues in the scan
+        const fileNode: FileTreeNode | undefined = this._treesManager.issuesTreeDataProvider.getFileIssuesTree(document.uri.fsPath);
+        if (fileNode instanceof CodeFileTreeNode) {
+            let diagnostics: vscode.Diagnostic[] = context.diagnostics.filter(
+                diagnostic => this.isJFrogSource(diagnostic.source) && diagnostic.range.contains(range)
```

Comment:
Let's split the AND operation for better readability

Reply:
this is a filter condition how can I split it? it looks clear to me. I have added comment above it to clarify